src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -118,3 +128,34 @@ def prep_null_label_for_url_encoding(label=None):
         label = '"{0}"'.format(label)
         label = ast.literal_eval(label)
     return label
+
+
+def get_appconfig_data_client(cmd, name, connection_string, auth_mode, endpoint):
+    azconfig_client = None
+    if auth_mode == "key":
+        connection_string = resolve_connection_string(cmd, name, connection_string)
+        azconfig_client = AzureAppConfigurationClient.from_connection_string(connection_string=connection_string,
+                                                                             user_agent=HttpHeaders.USER_AGENT)
+    if auth_mode == "login":
+        if not endpoint:
+            try:
+                if name:
+                    _, endpoint = resolve_resource_group(cmd, name)
+                else:
+                    raise CLIError("App Configuration endpoint or name should be provided if auth mode is 'login'.")
+            except Exception as ex:
+                raise CLIError("Failed to retrieve App Configuration endpoint from store name.\n" + str(ex))
+
+        from azure.cli.core._profile import Profile
+        profile = Profile(cli_ctx=cmd.cli_ctx)
+        # Due to this bug in get_login_credentials: https://github.com/Azure/azure-cli/issues/15179,
+        # we need to manage the AAD scope by passing appconfig endpoint as resource
+        cred, _, _ = profile.get_login_credentials(resource=endpoint)
+        azconfig_client = AzureAppConfigurationClient(credential=cred,
+                                                      base_url=endpoint,
+                                                      user_agent=HttpHeaders.USER_AGENT)
+
+    if not azconfig_client:
+        raise CLIError("Could not get App Configuration client due to insufficient permissions.")
```

Comment:
This check is misleading. AzureAppConfigurationClient will throw exceptions if it can't create the client.

Reply:
Done